src/compas/rpc/dispatcher.py

```diff
@@ -86,13 +86,12 @@ def _dispatch(self, name, args):
             * `'profile'` : A profile of the function execution.
 
         """
-        odict = {
-            'data': None,
-            'error': None,
-            'profile': None
-        }
+        odict = {"data": None, "error": None, "profile": None}
 
-        parts = name.split('.')
+        if len(args) > 1:
+            sys.path.insert(0, args[1])
```

Comment:
Add check otherwise it will insert the path on every function call:
```suggestion
            if args[1] not in sys.path:
                sys.path.insert(0, args[1])
```